buildSrc/src/main/java/net/neoforged/neodev/NeoDevPlugin.java

```diff
@@ -0,0 +1,527 @@
+package net.neoforged.neodev;
+
+import net.neoforged.minecraftdependencies.MinecraftDependenciesPlugin;
+import net.neoforged.moddevgradle.internal.NeoDevFacade;
+import net.neoforged.moddevgradle.tasks.JarJar;
+import net.neoforged.neodev.installer.CreateArgsFile;
+import net.neoforged.neodev.installer.CreateInstallerProfile;
+import net.neoforged.neodev.installer.CreateLauncherProfile;
+import net.neoforged.neodev.installer.InstallerProcessor;
+import net.neoforged.neodev.utils.DependencyUtils;
+import net.neoforged.nfrtgradle.CreateMinecraftArtifacts;
+import net.neoforged.nfrtgradle.DownloadAssets;
+import net.neoforged.nfrtgradle.NeoFormRuntimePlugin;
+import net.neoforged.nfrtgradle.NeoFormRuntimeTask;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository;
+import org.gradle.api.file.Directory;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.plugins.BasePluginExtension;
+import org.gradle.api.plugins.JavaPluginExtension;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Sync;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.api.tasks.bundling.AbstractArchiveTask;
+import org.gradle.api.tasks.bundling.Jar;
+import org.gradle.api.tasks.bundling.Zip;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class NeoDevPlugin implements Plugin<Project> {
+    static final String GROUP = "neoforge development";
+    static final String INTERNAL_GROUP = "neoforge development/internal";
+
+    @Override
+    public void apply(Project project) {
+        project.getPlugins().apply(MinecraftDependenciesPlugin.class);
+
+        var dependencyFactory = project.getDependencyFactory();
+        var tasks = project.getTasks();
+        var neoDevBuildDir = project.getLayout().getBuildDirectory().dir("neodev");
+
+        var rawNeoFormVersion = project.getProviders().gradleProperty("neoform_version");
+        var fmlVersion = project.getProviders().gradleProperty("fancy_mod_loader_version");
+        var minecraftVersion = project.getProviders().gradleProperty("minecraft_version");
+        var neoForgeVersion = project.provider(() -> (String) project.getVersion()); // TODO: is this correct?
```

Comment:
I believe you're meant to `toString` it and not cast it as it can be set to a non-string object.